packages/cbioportal-clinical-timeline/src/Timeline.tsx

```diff
@@ -203,14 +250,31 @@ const Timeline: React.FunctionComponent<ITimelineProps> = observer(function({
         zoomSelectBox: useRef(null),
         zoomSelectBoxMask: useRef(null),
         cursorText: useRef(null),
+        hoverStyleTag: useRef(null),
+        timelineViewPort: useRef(null),
     };
 
+    const memoizedHoverCallback = useCallback(
+        (e: React.MouseEvent) => {
+            hoverCallback(e, refs.hoverStyleTag);
+        },
+        [refs.hoverStyleTag]
+    );
+
     // on mount, there will be no element to measure, so we need to do this on equivalent
     // of componentDidMount
     useEffect(() => {
         setTimeout(() => {
-            store.updateViewPortWidth();
+            store.viewPortWidth = jQuery(
+                refs.timelineViewPort.current!
+            ).width()!;
         }, 10);
+
+        const cleanupDomEvents = bindToDOMEvents(store, refs);
+
+        return function cleanup() {
```

Comment:
is this what react calls on unmount?

Reply:
yes.  whatever function is returned from useEffect is called on unmount